libs/langchain/langchain/llms/mlflow.py

```diff
@@ -0,0 +1,90 @@
+from __future__ import annotations
+
+from typing import Any, Dict, List, Mapping, Optional
+
+from langchain_core.pydantic_v1 import BaseModel, Extra
+
+from langchain.callbacks.manager import CallbackManagerForLLMRun
+from langchain.llms.base import LLM
+
+
+# Ignoring type because below is valid pydantic code
+# Unexpected keyword argument "extra" for "__init_subclass__" of "object"
+class Params(BaseModel, extra=Extra.allow):  # type: ignore[call-arg]
+    """Parameters for MLflow"""
+
+    temperature: float = 0.0
+    n: int = 1
+    stop: Optional[List[str]] = None
+    max_tokens: Optional[int] = None
+
+
+class Mlflow(LLM):
+    """Wrapper around completions LLMs in MLflow.
+
+    To use, you should have the `mlflow[genai]` python package installed.
+    For more information, see https://mlflow.org/docs/latest/llms/deployments/server.html.
+
+    Example:
+        .. code-block:: python
+
+            from langchain.llms import Mlflow
+
+            completions = Mlflow(
+                target_uri="<target_uri>",
+                endpoint="<endpoint>",
+                params={
+                    "temperature": 0.1
+                }
+            )
+    """
+
+    endpoint: str
+    target_uri: str
+    params: Optional[Params] = None
+    client: Any = None
+
+    def __init__(self, **kwargs: Any):
+        super().__init__(**kwargs)
+        try:
+            from mlflow.deployments import get_deploy_client
+
+            self.client = get_deploy_client(self.target_uri)
+        except ImportError as e:
+            raise ImportError(
+                "Failed to create the client. "
+                "Please install mlflow with `pip install mlflow[genai]`."
+            ) from e
+
+    @property
+    def _default_params(self) -> Dict[str, Any]:
+        params: Dict[str, Any] = {
+            "target_uri": self.target_uri,
+            "endpoint": self.endpoint,
+            **(self.params.dict() if self.params else {}),
+        }
+        return params
+
+    @property
+    def _identifying_params(self) -> Mapping[str, Any]:
+        return self._default_params
+
+    def _call(
+        self,
+        prompt: str,
+        stop: Optional[List[str]] = None,
+        run_manager: Optional[CallbackManagerForLLMRun] = None,
+        **kwargs: Any,
+    ) -> str:
+        data: Dict[str, Any] = {
+            "prompt": prompt,
+            **(self.params.dict() if self.params else {}),
+        }
+        if s := (stop or (self.params.stop if self.params else None)):
+            data["stop"] = s
+        resp = self.client.predict(endpoint=self.endpoint, inputs=data)
+        return resp["choices"][0]["text"]
+
+    @property
+    def _llm_type(self) -> str:
+        return "mlflow"
```

Comment:
Please also add this new llm type to [`get_type_to_cls_dict` in `llms/__init__.py`](https://github.com/langchain-ai/langchain/blob/686162670e2fe15fb906999da84d36a273b2f25e/libs/langchain/langchain/llms/__init__.py#L761) so that it can be correctly serialized/unserialized using LangChain.

Test if it works using:
```python
import mlflow
from langchain.chains import LLMChain
from langchain.prompts import PromptTemplate
from langchain.llms import MLflow

completions = Mlflow(
    target_uri="<target_uri>",
    endpoint="<endpoint>",
    params={
        "temperature": 0.1
    }
)

llm_chain = LLMChain(
    llm=completions,
    prompt=PromptTemplate(input_variables=["adjective"], template="Tell me a {adjective} joke"),
)

saved_path = "./chain.yaml"
llm_chain.save(saved_path)
loaded_chain = load_chain(saved_path)
```

Reply:
@freemso The following code worked fine:

```python
import mlflow
from langchain.chains import LLMChain
from langchain.prompts import PromptTemplate
from langchain.llms import Mlflow
from langchain.chains.loading import load_chain

completions = Mlflow(
    target_uri="http://localhost:5000",
    endpoint="completions",
    params={"temperature": 0.1},
)

llm_chain = LLMChain(
    llm=completions,
    prompt=PromptTemplate(
        input_variables=["adjective"], template="Tell me a {adjective} joke"
    ),
)

saved_path = "./chain.yaml"
llm_chain.save(saved_path)
loaded_chain = load_chain(saved_path)
print(loaded_chain("funny"))
```

Output:

```
{'adjective': 'funny', 'text': "Sure, here's a classic one for you:\n\nWhy don't scientists trust atoms?\n\nBecause they make up everything!"}
```

Reply:
Awesome! Thanks!